bpdm-orchestrator-api/src/main/kotlin/org/eclipse/tractusx/orchestrator/api/model/TaskProcessingStateDto.kt

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.eclipse.tractusx.orchestrator.api.model
+
+import io.swagger.v3.oas.annotations.media.Schema
+import java.time.Instant
+
+@Schema(description = "Contains detail information about the current processing state of a cleaning task")
+data class TaskProcessingStateDto(
+    @get:Schema(description = "The last cleaning step this cleaning task has entered", required = true)
+    val cleaningStep: CleaningStep,
```

Comment:
What happens with `cleaningStep` and `reservationState` when cleaning is complete?

Reply:
I tried to catch that information in the schema descriptions. For the success state it makes sense to set those fields to null, but in case of an error it would be good to know the last step in which the error occurred. This way the fields are meant to read as "latestCleaningStep" and "latestReservationState"